### PR TITLE
Fix quick access new product

### LINKED ISSUE #28673: BO - Feature flag is enabled - Add new product from the quick access is NOK
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Previously reported here: https://github.com/PrestaShop/PrestaShop/pull/28594#pullrequestreview-993977071

The Add new product from the quick access is NOK when `feature flag is enabled`

### Expected behavior

Opens the modal of the product type when the user clicks on 'Add new product'. 
When the type is selected, the user is redirected to the product page with product type selected.

### Steps to reproduce

**Second issue**

**_Case1_**
1. Go to BO > Advanced Parameters > New & Experimental Features page
2. Enable `New product page - Single store` and save
3. In the Quick access > click on New product
4. See exception 

image

I attached a screen record

https://user-images.githubusercontent.com/16067358/171688808-581155bd-6fe2-4584-a76c-5474ca1094fe.mp4

**_Case2_**

1. Enable the multistore feature
2. Go to BO > Advanced Parameters > New & Experimental Features page
3. Enable `New product page - Multi store` and save
4. In the Quick access > click on New product
5. See exception 

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.2, 8.1

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by margud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Quick access fix with new product v2 page. Need to check new and old controllers.
| Type?             | bug fix 
| Category?         | BO
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28673 .
| How to test?      | First you need to upgrade default and new theme. Then enable new product page feature and try to create new product using quick access. Same cases should be tested when multi-shop is enabled.
| Possible impacts? | Updated BO JS and CSS so it might have some impact.

#### Screenshots

Old controller: 
image
New controller: 
image

### CODE REVIEW

File: admin-dev/themes/default/template/header.tpl [hunk 1]
```diff
@@ -112,7 +112,9 @@
 </head>
 
 {if $display_header}
-  <body class="lang-{$iso_user}{if $lang_is_rtl} lang-rtl{/if} ps_back-office{if $employee->bo_menu} page-sidebar{if $collapse_menu} page-sidebar-closed{/if}{else} page-topbar{/if} {$controller_name|escape|strtolower}">
+< class="lang-{$iso_user}{if $lang_is_rtl} lang-rtl{/if} ps_back-office{if $employee->bo_menu} page-sidebar{if $collapse_menu} page-sidebar-closed{/if}{else} page-topbar{/if} {$controller_name|escape|strtolower}"
```
Comment (nicosomb) [inline admin-dev/themes/default/template/header.tpl]: the `body` tag has disappear.

File: classes/QuickAccess.php [hunk 2]
```diff
@@ -105,6 +108,13 @@ public static function getQuickAccessesWithToken($idLang, $idEmployee)
                     $tokenString = $admin_tab[1] . (int) Tab::getIdFromClassName($admin_tab[1]) . $idEmployee;
                 }
                 $quickAccess[$index]['link'] = Context::getContext()->link->getBaseLink() . basename(_PS_ADMIN_DIR_) . '/' . $quick['link'];
+                if ($quick['link'] === 'index.php/sell/catalog/products/new' || $quick['link'] === 'index.php/sell/catalog/products-v2/create') {
```
Comment (nicosomb) [inline classes/QuickAccess.php]: I think that these hardcoded values should be in constants.

File: classes/QuickAccess.php [hunk 3]
```diff
@@ -105,6 +108,13 @@ public static function getQuickAccessesWithToken($idLang, $idEmployee)
...
+                    //if new product page feature is enabled we create new product v2 modal popup
+                    if (self::productPageV2Enabled()) {
+                        $quickAccess[$index]['link'] = Context::getContext()->link->getAdminBaseLink() . basename(_PS_ADMIN_DIR_) . '/' . 'index.php/sell/catalog/products-v2/create';
+                        $quickAccess[$index]['class'] = 'new-product-button';
```
Comment (nicosomb) [inline classes/QuickAccess.php:125]: Should we add css class on PHP side?

File: classes/QuickAccess.php [hunk 4]
```diff
@@ -135,4 +145,18 @@ public function toggleNewWindow()
...
+     */
+    private static function productPageV2Enabled(): bool
+    {
+        $multistoreFeature = SymfonyContainer::getInstance()->get('prestashop.adapter.multistore_feature');
+
+        if (!$multistoreFeature->isActive()) {
+            return SymfonyContainer::getInstance()->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_PRODUCT_PAGE_V2);
+        }
+
+        return SymfonyContainer::getInstance()->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_PRODUCT_PAGE_V2_MULTI_SHOP);
```
Comment (nicosomb) [inline classes/QuickAccess.php]: Maybe you can remove one `return` with this: 

```
return SymfonyContainer::getInstance()->get('prestashop.core.admin.feature_flag.repository')->isEnabled($multistoreFeature->isActive() ? FeatureFlagSettings::FEATURE_FLAG_PRODUCT_PAGE_V2_MULTI_SHOP : FeatureFlagSettings::FEATURE_FLAG_PRODUCT_PAGE_V2_MULTI_SHOP);
```

File: classes/QuickAccess.php (hunk 3 above)
Comment (margud) [reply classes/QuickAccess.php:125]: well the quick select is generated here, i can't think of any other way of adding class that could help us find specific selection option.

**Review by nicosomb — DISMISSED** [review]

**Review by nicosomb — APPROVED** [review]

File: classes/QuickAccess.php (hunk 3 above)
Comment (jolelievre) [reply classes/QuickAccess.php:125]: Indeed it's the only place where we can customize the quick links, and it's the same principle as defining some classes in FormType actually.

Anyway, this is a temporary code, when the v2 is officially released and the former one is removed we'll be able to clean all this code.

**Review by jolelievre — APPROVED** [review]
Thanks @margud

**Review by zuk3975 — APPROVED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hello @margud,

**Issue1**

1. Go to BO > Advanced Parameters > New & Experimental Features page & Enable New product page - Single store
2. Click on the "Quick access" > Manage your quick access
3. Enable "New window" for the "New product" 
image
4. Click on "Quick access" > Add a new product
5. Click on "Add new product" button => the add new product page is displayed in the same page

It should be opened on another page


https://user-images.githubusercontent.com/16067358/198686293-9c9cacc5-cff7-4500-a754-303fc7453b4d.mp4


ping @MatShir, what do you think?

**Issue2**

1. Go to BO > Shop Parameters > General page 
2. Enable multistore
3. Create a new shop & configure it
4. Go to BO > Advanced Parameters > New & Experimental Features page & Enable New product page - Multi store
5. Select all shops in the header
6. Click on "Quick access" > Add a new product
7. An alert `"This page is only compatible in a single store context. To access the page, please select a store or "disable the multistore feature"` is displayed
image
8. Click on the "disable the multistore feature" link
9. See error the General page is displayed in the modal & we cannot scroll to disable multistore

I attached a screen record

https://user-images.githubusercontent.com/16067358/198682515-342ed412-7ab2-4d26-acba-907a356fc295.mp4

**Issue3**

1. Go to BO > Advanced Parameters > Team > Profiles page
2. Create a new profile
3. This profile has only "View" 
image
4. Create an employee with this profile & try to sign in with it 
image
5. Click on "Quick access" > Add a new product
6. See error => No alert displayed 

https://user-images.githubusercontent.com/16067358/198684942-bf84d054-e74d-48e5-ac34-9b9d65883d8f.mp4

Thanks!

**Comment by jolelievre** [comment]
HI @khouloudbelguith 

thanks for the feedback here's my opinion on the bugs you found

**Issue 1**
We had no idea that the quick link had such a feature to allow open it in a new tab, the creation process for a product is completely different from what was done previously and needs to pass through this modal. In this case I don't know what would be the best solution between those two:
- open the type selection in another tab instead of the modal
- open the modal but open the edition form in another tab after the type has been selected

I think @MatShir should decide the best solution here, but in any case I don't think this should block the current PR, since this kind of configuration is kind of an edge case. Besides this PR is blocking for another feature (see issue 2 below 😉)

**Issue 2**
The creation process for multishop has not been implemented yet, so it can't be a blocker for this PR which was focused on single shop. And actually, there is a PR to handle this all correctly https://github.com/PrestaShop/PrestaShop/pull/27712
But it's blocked for now, because this PR modifies common components so I need this one to be merged to finalize the multishop one

**Issue 3**
This weird behaviour is actually due to our poor management of permissions in the BO, especially when someone doesn't have the right for a specific page the default behaviour is to redirect to the dashboard... and that's it! No error message explaining what happened, no default forbidden page to display. This is a global problem not related to this particular feature.
Besides, I admit the UX is far from ideal here, but it still does the job and prevents an Employee without the right access to create a product. But the real fix should rather be not to display the quick link but here again it's another issue.


So TLDR;
**Issue 1**: should be handled in a dedicated issue (although I'm not sure it's the priority)
**Issue 2**: the feature will have to be tested in this PR https://github.com/PrestaShop/PrestaShop/pull/27712 but first we need this one merged
**Issue 3**: Either improve our permission management globally in the BO, or handle a quick/ugly fix specific to this use case, but in both case, it should be decided (and fixed) in another issue

Again, that's just my opinion I just wanted to bring all the information, it's up to @MatShir to decide anyway 🙂

**Comment by MatShir** [comment]
I agree with @jolelievre. Thanks for bring to light this issues @khouloudbelguith. 
If it is ok and there is no other issues in the PR, let's merge the PR and I will create the related issue. WDYT @khouloudbelguith ?

**Review by khouloudbelguith — APPROVED** [review]
Hi @margud, @jolelievre,

It is ok :heavy_check_mark: 
I checked "Add new product" from quick access with all BO pages.
I tried with iPad => ok
I tried with multistore enabled & disabled => OK
I tried with Arabic language => Ok

https://user-images.githubusercontent.com/16067358/199714968-ab9bbf7d-5a29-4525-82e3-ffdccd4212d3.mp4

image

Thanks!

**Comment by khouloudbelguith** [comment]
Hi @margud, @jolelievre, @MatShir,

I created the issues
https://github.com/PrestaShop/PrestaShop/issues/30202
https://github.com/PrestaShop/PrestaShop/issues/30201
https://github.com/PrestaShop/PrestaShop/issues/30200

Thank you all!

**Comment by jolelievre** [comment]
Thanks @margud and @khouloudbelguith
